pkg/client/build_test.go

```diff
@@ -1708,6 +1710,92 @@ func testBuild(t *testing.T, when spec.G, it spec.S) {
 			})
 		})
 
+		when("Extensions option", func() {
+			it.Before(func() {
+				subject.experimental = true
+				defaultBuilderImage.SetLabel("io.buildpacks.buildpack.order-extensions", `[{"group":[{"id":"extension.1.id","version":"extension.1.version"}]}, {"group":[{"id":"extension.2.id","version":"extension.2.version"}]}]`)
+				defaultWindowsBuilderImage.SetLabel("io.buildpacks.buildpack.order-extensions", `[{"group":[{"id":"extension.1.id","version":"extension.1.version"}]}, {"group":[{"id":"extension.2.id","version":"extension.2.version"}]}]`)
+			})
+
+			assertOrderEquals := func(content string) {
+				t.Helper()
+
+				orderLayer, err := defaultBuilderImage.FindLayerWithPath("/cnb/order.toml")
+				h.AssertNil(t, err)
+				h.AssertOnTarEntry(t, orderLayer, "/cnb/order.toml", h.ContentEquals(content))
+			}
+
+			it("builder order-extensions is overwritten", func() {
+				additionalEx := ifakes.CreateExtensionTar(t, tmpDir, dist.ExtensionDescriptor{
+					WithAPI: api.MustParse("0.3"),
+					WithInfo: dist.ModuleInfo{
+						ID:      "extension.add.1.id",
+						Version: "extension.add.1.version",
+					},
+				})
+
+				h.AssertNil(t, subject.Build(context.TODO(), BuildOptions{
+					Image:      "some/app",
+					Builder:    defaultBuilderName,
+					ClearCache: true,
+					Extensions: []string{additionalEx},
+				}))
+				h.AssertEq(t, fakeLifecycle.Opts.Builder.Name(), defaultBuilderImage.Name())
+
+				assertOrderEquals(`[[order]]
+
+  [[order.group]]
+    id = "buildpack.1.id"
+    version = "buildpack.1.version"
+
+[[order]]
+
+  [[order.group]]
+    id = "buildpack.2.id"
+    version = "buildpack.2.version"
+
+[[order-extensions]]
+
+  [[order-extensions.group]]
+    id = "extension.add.1.id"
+    version = "extension.add.1.version"
+`)
+			})
+
+			when("id - no version is provided", func() {
+				it("resolves version", func() {
+					h.AssertNil(t, subject.Build(context.TODO(), BuildOptions{
+						Image:      "some/app",
+						Builder:    defaultBuilderName,
+						ClearCache: true,
+						Extensions: []string{"extension.1.id"},
+					}))
+					h.AssertEq(t, fakeLifecycle.Opts.Builder.Name(), defaultBuilderImage.Name())
+
+					assertOrderEquals(`[[order]]
+
+  [[order.group]]
+    id = "buildpack.1.id"
+    version = "buildpack.1.version"
+
+[[order]]
+
+  [[order.group]]
+    id = "buildpack.2.id"
+    version = "buildpack.2.version"
+
+[[order-extensions]]
+
+  [[order-extensions.group]]
+    id = "extension.1.id"
+    version = "extension.1.version"
+`)
+				})
+			})
+		})
+
+		//TODO: "all buildpacks are added to ephemeral builder" test after extractPackaged() is completed.
```

Comment:
Ah - looking at your comment [here](https://github.com/buildpacks/pack/pull/1651#issuecomment-1465596855), I think we can exclude this one, as there is no such thing as a "meta extension".

Reply:
Ohhh, great!